Algorithm/QCAlgorithm.Python.cs

```diff
@@ -52,56 +52,130 @@ public void SetPandasConverter()
         /// </summary>
         /// <param name="type">Data source type</param>
-        /// <param name="symbol">Key/Symbol for data</param>
+        /// <param name="underlying">The underlying symbol for the custom data</param>
+        /// <param name="resolution">Resolution of the Data Required</param>
+        /// <param name="timeZone">Specifies the time zone of the raw data</param>
+        /// <param name="fillDataForward">When no data available on a tradebar, return the last data that was generated</param>
+        /// <param name="leverage">Custom leverage per security</param>
+        /// <returns>The new <see cref="Security"/></returns>
+        public Security AddData(PyObject type, Symbol underlying, Resolution resolution, DateTimeZone timeZone, bool fillDataForward = false, decimal leverage = 1.0m, object _ = null, object __ = null, object ___ = null)
+        {
+            return AddData(CreateType(type), underlying, resolution, timeZone, fillDataForward, leverage);
+        }
+
+        /// <summary>
+        /// AddData a new user defined data source, requiring only the minimum config options.
+        /// </summary>
+        /// <param name="dataType">Data source type</param>
+        /// <param name="ticker">Key/Ticker for data</param>
         /// <param name="resolution">Resolution of the Data Required</param>
         /// <param name="timeZone">Specifies the time zone of the raw data</param>
         /// <param name="fillDataForward">When no data available on a tradebar, return the last data that was generated</param>
         /// <param name="leverage">Custom leverage per security</param>
         /// <returns>The new <see cref="Security"/></returns>
-        public Security AddData(PyObject type, string symbol, Resolution resolution, DateTimeZone timeZone, bool fillDataForward = false, decimal leverage = 1.0m)
+        public Security AddData(Type dataType, string ticker, Resolution resolution, DateTimeZone timeZone, bool fillDataForward = false, decimal leverage = 1.0m)
         {
-            return AddData(CreateType(type), symbol, resolution, timeZone, fillDataForward, leverage);
+            var baseInstance = dataType.GetBaseDataInstance();
+            if (!baseInstance.RequiresMapping())
+            {
+                var symbol = new Symbol(
+                    SecurityIdentifier.GenerateBase(dataType, ticker, Market.USA, dataType.GetBaseDataInstance().RequiresMapping()),
+                    ticker);
+                return AddDataImpl(dataType, symbol, resolution, timeZone, fillDataForward, leverage);
+            }
+            // If we need a mappable ticker and we can't find one in the SymbolCache, throw
+            Symbol underlying;
+            if (!SymbolCache.TryGetSymbol(ticker, out underlying))
+            {
+                throw new InvalidOperationException("Requires mapping but passed ticker which is not in the cache");
```

Comment:
I think we can further improve this error message, maybe something like: `The custom data type {dataType.Name} requires mapping but the provided ticker is not in the cache. Please add this custom data type using the related underlying Symbol instance or perform this call after the underlying Security has been added using AddSecurity() methods. An example use case can be found at {nameof(RegressionAlgorithm)}`. We want the users to understand and be able to fix the exception by themselves if possible.

Reply:
Although, you won't be able to use `nameof(RegressionAlgorithm)` so a string literal will suffice (since its in the `Algorithm.CSharp` which isn't referenced via `Algorithm`)